### PR TITLE
Migrate information from dev book

### LINKED ISSUE #2790: Migrate standalone docs to rustdoc
Docs written in #2732 should be transformed and migrated to regular rustdocs.

### PR DESCRIPTION
**Pull request by marc-casperlabs**
This PR should migrate most of the usable information from #2732.

Closes #2790.

### CODE REVIEW

File: node/src/components.rs [hunk 1]
```diff
@@ -1,7 +1,49 @@
...
+//!
+//! A typical cycle for components is to receive an event, either originating from the outside, or
+//! as the result of an effect created by the component. This event is processed in the
+//! [`handle_event`](Component::handle_event) function, potentially returning effects that may
+//! produce new events.
+//!
+//! # Error and halting states
+//!
+//! Components in general are expected to be able to handle every input (that is every
+//! [`Component::Event`]) in every state. Unexpected inputs should usually be logged and discard, if
```
Comment (rafal-ch) [inline node/src/components.rs]:
```suggestion
//! [`Component::Event`]) in every state. Unexpected inputs should usually be logged and discarded, if
```

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

File: node/src/components.rs (hunk 1 above)
Comment (marc-casperlabs) [reply node/src/components.rs]: Have to fix this manually due to the linebreak :)

**Pull request merged**